### PR TITLE
Fix issues with secondary views and layer management.

### PR DESCRIPTION
**Pull request by asajeffrey**
Fix issues with secondary views and layer management.

### CODE REVIEW

File: webxr/openxr/mod.rs [hunk 1]
```diff
@@ -895,10 +896,8 @@ impl OpenXrDevice {
                 })?[0];
 
             let secondary_extent = Extent2Di {
-                width: (view_configuration.recommended_image_rect_width / SECONDARY_VIEW_DOWNSCALE)
-                    as i32,
-                height: (view_configuration.recommended_image_rect_height
-                    / SECONDARY_VIEW_DOWNSCALE) as i32,
+                width: view_configuration.recommended_image_rect_width as i32,
```
Comment (Manishearth) [inline webxr/openxr/mod.rs:899]: why did we move the downscaling?

File: webxr/openxr/mod.rs [hunk 2]
```diff
@@ -1103,16 +1106,44 @@ impl DeviceAPI for OpenXrDevice {
...
         }
 
-        let mut guard = self.shared_data.lock().unwrap();
-        let mut data = guard.as_mut().unwrap();
-        let frame_state = match self.frame_waiter.wait() {
-            Ok(frame_state) => frame_state,
-            Err(e) => {
-                error!("Error waiting on frame: {:?}", e);
-                return None;
+        let (frame_state, secondary_state) = if self
```
Comment (Manishearth) [inline webxr/openxr/mod.rs:1109]: i wonder if a `secondary_enabled` can be stored on `self` instead: it's readonly

Comment (asajeffrey) [reply webxr/openxr/mod.rs:1109]: Problem is it's needed by the `views()` function, so we'd end up duplicating it.

Comment (Manishearth) [reply webxr/openxr/mod.rs:1109]: Yes, but it's a boolean, so it's cheap and better than locking twice

File: webxr/openxr/mod.rs (hunk 1 above)
Comment (asajeffrey) [reply webxr/openxr/mod.rs:899]: Keeping the original value around seems cleaner, e.g. different layer types can use different scaling. But it's an inessential change, we can go back if we like.

File: webxr/openxr/mod.rs (hunk 2 above)
Comment (asajeffrey) [reply webxr/openxr/mod.rs:1109]: That's true, this double-locking is annoying.

Comment (asajeffrey) [reply webxr/openxr/mod.rs]: Fixed.

File: webxr/openxr/mod.rs (hunk 1 above)
Comment (Manishearth) [reply webxr/openxr/mod.rs:900]: Ah, seems fine.

**Pull request merged**